src/DssSpell.sol

```diff
@@ -112,6 +125,17 @@ contract DssSpellAction is DssAction {
             ARB_GAS_PRICE_BID,
             ARB_MAX_SUBMISSION_COST
         );
+
+        // ------------------ Pause Starknet Goerli L2DaiTeleportGateway -----------------
+        // Forum: https://forum.makerdao.com/t/community-notice-pecu-to-redeploy-teleport-l2-gateways/19550
+        // L2 Spell to execute via STARKNET_GOV_RELAY:
+        // src: https://github.com/makerdao/starknet-spells-goerli/blob/b7ca995cf1d266aa2382d85e35a86b4fae52aa15/src/spell.cairo
+        // contract: https://testnet.starkscan.co/class/0x00a052591661d7e249b46a1084c63b14dae6aa8b1a56ab3f7df8c8add1c374b1#overview
+        StarknetGovRelayLike(STARKNET_GOV_RELAY).relay{value: STA_L1_CALL_VALUE}(STARKNET_L2_SPELL);
```

Comment:
Mmm even if STARKNET_GOV_RELAY pre-receives ETH through `selfdestruct`, the actual [call](https://goerli.etherscan.io/address/0x8919aefA417745F22c6af5AD6550E83159a373F3#code#F1#L49) it makes to `starknet` is passing `msg.value`, which would be 0. 

So if `starknet` expects eth I think the options are:
1. Pre-fund the pause proxy using `selfdestruct` as already suggested (and seems like the current code supports).
2. Add an authed intermediate contract between the pause proxy and the gov relay and fund it normally.
3. Redeploy the governance relays on both sides, adding a default payable function on l1.

IMO option 3 is the least hacky and also aligns with the Arbitrum intended scheme.

Reply:
Redeployment has to go through current STARKNET_GOV_RELAY so it would need to use option 1 or 2 anyway. 
There is a large upgrade planned on `starknet` in next few months. It will make us to redeploy STARKNET_GOV_RELAY. It will be a good moment to upgrade it without any extra work. Until then I would use #1 or #2.

Reply:
Option 1 seems like the way to go for this one, as @maciejka mentionned we will have to rewrite it anyway in the short term future.

Reply:
We decided to go for option 1, which does not require code changes.
@olivdb I think this can be resolved.